palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/request/AuditMessage.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.palisade.request;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.exception.PalisadeRuntimeException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+
+/**
+ * This is the parent class for Audit information.  It represents the common component of the data that is to be
+ * sent to audit service.
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AuditMessage {
+
+    protected static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @JsonProperty("userId")
+    protected final String userId; //Unique identifier for the user.
+
+    @JsonProperty("resourceId")
+    protected final String resourceId;  //Resource that that is being asked to access.
+
+    @JsonProperty("context")
+    protected final Context context;   //Relevant context information about the request.
+
+    @JsonProperty("timestamp")
+    protected final String timestamp;  //when the message was created
+
+    @Value("${spring.application.name}")
```

Comment:
suspect this shouldn't be needed as the serviceName will always be "palisade-service" here and should not be the name of the actual variant of palisade service (E.g SimplePalisadeService)

Reply:
similar comment made in: 
https://github.com/gchq/Palisade-services/pull/146#discussion_r459588740